core/arch/arm/sm/std_smc.c

```diff
@@ -41,6 +41,67 @@ static const TEE_UUID uuid = {
 	{0x98, 0xd2, 0x74, 0xf4, 0x38, 0x27, 0x98, 0xbb},
 };
 
+#include <trace.h>
+
+static unsigned long arm_arch_version(void)
+{
+	return SMCCC_V_1_1;
+}
+
+static unsigned long arm_arch_feature(unsigned long a1)
+{
+	unsigned long func_id = a1;
+
+	switch (func_id) {
+	case ARM_SMCCC_VERSION:
+		return ARM_SMCCC_RET_SUCCESS;
+	default:
+		return ARM_SMCCC_RET_NOT_SUPPORTED;
+	}
+}
+
+static unsigned long __weak arm_arch_soc_id(void)
+{
+	return ARM_SMCCC_RET_NOT_SUPPORTED;
+}
+
+static unsigned long __weak arm_arch_workaround_1(void)
+{
+	return ARM_SMCCC_RET_NOT_REQUIRED;
+}
+
+static unsigned long __weak arm_arch_workaround_2(void)
+{
+	return ARM_SMCCC_RET_NOT_REQUIRED;
+}
+
+void smc_arch_handler(struct thread_smc_args *args)
```

Comment:
This looks unnecessary complicated. Do we really expect platform specific extensions? If so, do we have any in the pipe? If not please simplify this function and use only a minimum of empty "helper" functions.

Reply:
ok. +1